### ISSUE TITLE
Wrong behaviour with multi-line tabs, single line tabs, etc.

### ISSUE THREAD

**Issue by e3roll** [notepad-plus-plus/notepad-plus-plus#2241]
Hi, I got three little problems. I do not know if they are bugs or if they are design choices.
### Description of the Issue

If I open many files, like 30, with the settings as shown in the picture, then when I try to drag or move one, it is hard to understand the behaviour so to move it to the right place. Plus, there is a visual glitch on the right line that divides the file and its document map.

![tabbar](https://cloud.githubusercontent.com/assets/7842271/18228677/15cd1e34-722f-11e6-9d9a-c0a38c903433.JPG)

Furthermore, suppose you arranged them as you wanted it, now try to close "all tabs to the right/left", it closes some random tabs. I guess, npp closes the tabs that were to the right/left at the time you opened all of them before re-arranging.

Moreover, if I use single line tab bar (multi-line disabled) and if I open, again, a lot of files, it is so difficult to rearrange them. You can't move in one single movement, let's say, the last tab to the first position. You have to move to the first position on that "page" (don't know how it's called), then go and click the arrows (like for scrolling, don't know how it's called), then move the tab again to the first position on the "page" and so on, untill it's the real first tab.
#### Steps to Reproduce the 1st Issue
1. Open many tabs, so as to have more than one line of tabs.
2. Try to arrange them as you want.
3. Weird glitches and confusing behaviour.
#### Steps to Reproduce the 2nd Issue
1. Right click, "close all to the left" or "close all to the right".
#### Steps to Reproduce the 3rd Issue
1. Open many tabs, so as to have more than one "page" of tabs.
2. Try moving the last tab to the absolute first position in one move.
### Expected Behavior

What I expected is:
1. To move tabs just like in Firefox or many other programs. But with the possibility to have multi-line tab bar and move any tab to any line, without that weird behaviour that's quickly moving tabs.
2. To close the tabs that are (at the moment I click) to the left/right.
3. To move tabs exactly like Firefox with an automatic scrolling.
### Suggestions or improvements

I would like two suggest two things:
- On a single-line tab bar configuration, with many tabs, to be able to scroll (like Firefox) with the mouse wheel and not to be forced to click in the little arrows on the right. (I am not talking anymore about moving tabs).
- In addition, on any configuration (single or multi-line tab bar) the possibility to right click a tab and to have the option to "move to the first" and "move to the last".
### Debug Information

Notepad++ v6.9.1
Build time : Mar 28 2016 - 19:48:40
Path : D:\Aplicaciones\Notepad++ 6.9.1\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 8.1
Plugins : ComparePlugin.dll mimeTools.dll NppConverter.dll NppExport.dll PluginManager.dll 
## 

I hope you can fix it or help me! Thanks in advance and for this awesome program.
Eugenio.

**Comment by kamilrymarz** [comment]
I agree that Drag&Drop is unusable when multiline is enabled. In current implementation tab is moved during mouse move  and row with selected tub must always be a bottom row. This two properties makes that when you drag item between rows then dragged item will move to other row and whole row will go down so cursor will point to different place than dragged item.  I see two possible fix for this:

1. Do not move row with selected item. I looked for such property in TabControl but I couldn't find anything so I'm not sure if it is possible with current implementation of TabBar.

2. Change implementation of Drag&Drop : tab hit by cursor during dragging will be marked as target and dragged tab will be moved only once at the end of Drag&Drop.

I found in code that currently tab is moved by shifting. In my opinion copy of moved item should be inserted on the proper position and moved item should be deleted from original position. This will have same effect and will

Also I think that in single line mode during Drag&Drop when cursor is to the left or right from tab  then TabBar should move to left or right.

In v6.9.2 you can move tab bar with scroll instead of clicking arrows.

**Comment by AngryGamer** [comment]
1st issue fixed in [9967c59](https://github.com/notepad-plus-plus/notepad-plus-plus/commit/9967c59912a05bfac57854e583d573877fdcb787)

**Comment by MetaChuh** [comment]
please update your notepad++ to the latest version [<kbd>Notepad++ 7.6.6</kbd>](https://notepad-plus-plus.org/download/v7.6.6.html) and migrate or reinstall your plugins if you update from version 7.5.9 or below.
if your issue still persists, and it is still of relevance to you, please feel free to file a new issue as easy to understand and reproducible as possible.

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)
